src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +224,428 @@ def _updateImage(self):
+                        # uses less memory. As the display size is small (compared
+                        # to the size of the input image, it shouldn't actually
+                        # affect much the output.
+                        logging.info("AR image is very large %s, will convert to "
+                                     "azimuthal projection in reduced precision.",
+                                     data_raw_bg_corr[key].shape)
+                        y, x = data_raw_bg_corr[key].shape
+                        if y > x:
+                            small_shape = 1024, int(round(1024 * x / y))
+                        else:
+                            small_shape = int(round(1024 * y / x)), 1024
+                        # resize
+                        data_raw_bg_corr[key] = img.rescale_hq(data_raw_bg_corr[key], small_shape)
+
+                # convert data into rectangular format (theta-phi-representation)
+                # Check if rectangular converted representation already was calculated for requested ebeam pos.
+                # Note: This calc is very time consuming. Takes about 3.6 sec for one ebeam pos (conversion of 6 images)
+                # and tested on an image of size (256, 1024)
+
+                if ebeam_pos not in polarimetry_cache_raw.keys():
+                    # Get the center wavelength of the filter used (no filter aka "pass-through" use fallback)
+                    # Does not matter from which of the 6 images as they all must have been recorded with the same
+                    # filter -> reuse key
+                    # TODO don't like this solution, key will be the last value of key in the for loop above
+                    # the value of key might be a different one each time...and it is only to get the wl of the filter.
+                    # not sure this is still working in python 3 as I learned today ;)
+                    fallback_wl = 650e-9
+                    band = data_raw_bg_corr[key].metadata.get(model.MD_OUT_WL, fallback_wl)
+                    if isinstance(band, tuple):  # wl is usually tuple of min/max value
+                        wl = sum(band) / len(band)
+                    else:  # handles if band is str
+                        # TODO if type is "str", support center wavelength based on color
+                        wl = fallback_wl
+
+                    # Warning: allocates lot of memory, which will not be free'd until
+                    # the current thread is terminated.
+
+                    # TODO allow variable input size? Calc based on raw data?
+                    size = (180, 720)  # (2*90, 2*360) defines the resolution of the displayed image
+
+                    for key in data_raw_bg_corr.keys():
+                        data_raw_bg_corr[key] = angleres.AngleResolved2Rectangular(data_raw_bg_corr[key], size)
+                    polarimetry_cache_raw[ebeam_pos] = data_raw_bg_corr  # dict containing the bg corrected raw data
```

Comment:
That `data_raw_bg_corr` doesn't seem ever reused. It's only there to know whether the polarimetry results are present or not. I'd suggest to modify a little bit the structure of `polarimetry_cache_raw` to be dict ((float,float) -> dict (str -> DataArray)): ebeam_pos -> {polarimetry -> stoke array}.

This would avoid to have two kinds of keys in the `polarimetry_cache_raw` dict, and also should avoid having to pass `ebeam_pos` to `calcPolarimetry()` (which is nice because that function shouldn't have to care).

Reply:
I understand that having to kinds of keys, is not nice and should be possible to change. However, I don't get how that should work with not passing the ebeam pos to `calcPolarimetry`.